tests/test_acasclient.py

```diff
@@ -2501,6 +2356,383 @@ def csv_to_txt(data_file_to_upload, dir):
+
+        # Verify that the restricted_experiment_code_name is in the linkedExperiments for the user
+        self.assertIn(restricted_experiment_code_name, [e['code'] for e in global_lot_dependencies['linkedExperiments']])
+
+        # Get the acls for the restricted experiment from the global_lot_dependencies
+        restricted_experiment_acls = [acl for acl in global_lot_dependencies['linkedExperiments'] if acl['code'] == restricted_experiment_code_name][0]['acls']
+        
+        # Verify the expected acls are read: true, write: true, delete: true
+        self.assertTrue(restricted_experiment_acls['read'])
+        self.assertTrue(restricted_experiment_acls['write'])
+        self.assertTrue(restricted_experiment_acls['delete'])
+
+    @requires_node_api
+    @requires_basic_cmpd_reg_load
+    def test_003_save_meta_lot(self):
+        """Test post meta lot."""
+
+        # Create a restricted project 
+        project = self.create_basic_project_with_roles()
+    
+        # Bulk load some compounds to we don't interfere with CMPD-0000001-001
+        self.basic_cmpd_reg_load(project.code_name)
+        all_lots = self.client.get_all_lots()
+
+        # Sort lots by id and get the latest corp id
+        # This is because we dont' get the corp id in the response from the bulkload
+        all_lots = sorted(all_lots, key=lambda lot: lot['id'])
+        restricted_project_lot_corp_name = all_lots[-1]['lotCorpName']
+        
+        # The default user is 'bob' and bob has cmpdreg admin role
+        # The basic cmpdreg load has a lot registered that is unrestricted (Global project)
+        meta_lot = self.client.\
+            get_meta_lot(restricted_project_lot_corp_name)
+        self.assertIsNotNone(meta_lot)
+        self.assertEqual(meta_lot['lot']['corpName'], restricted_project_lot_corp_name)
+
+        # Verify our cmpdreg admin user can save the restricted lot
+        meta_lot["lot"]["color"] = "red"
+        response = self.client.\
+            save_meta_lot(meta_lot)
+        self.assertEqual(len(response["errors"]), 0)
+        self.assertIn("metalot", response)
+        self.assertIn("lot", response["metalot"])
+        self.assertEqual(response["metalot"]["lot"]["color"], "red")
+
+        # Verify our admin user can update the project
+        meta_lot["lot"]["project"] = project.code_name
+        response = self.client.\
+            save_meta_lot(meta_lot)
+        self.assertEqual(len(response["errors"]), 0)
+        self.assertIn("metalot", response)
+        self.assertIn("lot", response["metalot"])
+        self.assertEqual(response["metalot"]["lot"]["project"], project.code_name)
+
+        # Now create a user that is not a cmpdreg admin and does not have access to the restricted project
+        user_client = self.create_and_connect_backdoor_user(acas_user=False, acas_admin=False, creg_user=True, creg_admin=False)
+              
+        # User should NOT be able fetch the restricted lot
```

Comment:
nit: is "fetch" the right word here? It seems like this code is attempting to edit / update the meta lot